doc/classes/TileMap.xml

```diff
@@ -5,8 +5,8 @@
 	</brief_description>
 	<description>
 		Node for 2D tile-based maps. Tilemaps use a [TileSet] which contain a list of tiles which are used to create grid-based maps. A TileMap may have several layers, layouting tiles on top of each other.
-		For performance reasons, all TileMap updates are batched at the end of a frame. Notably, this means that scene tiles from a [TileSetScenesCollectionSource] may be initialized after their parent.
-		To force an update earlier on, call [method update_internals].
+		For performance reasons, all TileMap updates are batched at the end of a frame. Notably, this means that scene tiles from a [TileSetScenesCollectionSource] may be initialized after their parent. This is only queued when inside the scene tree.
+		To force an update earlier on, or when not in the scene tree, call [method update_internals].
```

Comment:
Forced updates are useless when a TileMapLayer is outside the tree, so I don't think I would mention it.

Reply:
So the data updated is not relevant except for the physics/rendering/navigation/etc.? Not for the data accessible/usable by users? If so then this should be sufficient to work things out, and might not need forcing any updates on tree exit

Reply:
Yes. These updates are for runtime only, when the TileMap is in the tree. All subsystems updates just clear everything when TileMapLayer isn't in the tree. 

So those:
```
	_build_runtime_update_tile_data();

	// Update all subsystems.
	_rendering_update();
	_physics_update();
	_navigation_update();
	_scenes_update();
#ifdef DEBUG_ENABLED
	_debug_update();
#endif // DEBUG_ENABLED

	_clear_runtime_update_tile_data();	
```

Are forced to cleanup everything using something like:
```
	bool forced_cleanup = in_destructor || !enabled || !tile_set.is_valid() || !is_visible_in_tree();
```
or similar. (here, is_visible_in_tree ensure we clean up if the node is not in the tree)